statistics/handle/globalstats/global_stats.go

```diff
@@ -122,27 +121,21 @@ func MergePartitionStats2GlobalStats(
 			err = errors.Errorf("unknown physical ID %d in stats meta table, maybe it has been dropped", partitionID)
 			return
 		}
-
 		tableInfo := partitionTable.Meta()
 		var partitionStats *statistics.Table
-
-		partitionStats, ok = allPartitionStats[partitionID]
-		// If pre-load partition stats isn't provided, then we load partition stats directly and set it into allPartitionStats
-		if !ok {
-			var err1 error
-			partitionStats, err1 = loadTablePartitionStatsFn(tableInfo, &def)
-			if err1 != nil {
-				if skipMissingPartitionStats && types.ErrPartitionStatsMissing.Equal(err) {
-					globalStats.MissingPartitionStats = append(globalStats.MissingPartitionStats, fmt.Sprintf("partition `%s`", def.Name.L))
-					continue
-				}
-				err = err1
-				return
+		partitionStats, err1 := loadTablePartitionStatsFn(tableInfo, &def)
+		if err1 != nil {
+			if skipMissingPartitionStats && types.ErrPartitionStatsMissing.Equal(err) {
+				globalStats.MissingPartitionStats = append(globalStats.MissingPartitionStats, fmt.Sprintf("partition `%s`", def.Name.L))
+				continue
 			}
-			allPartitionStats[partitionID] = partitionStats
+			err = err1
+			return
 		}
 
 		for i := 0; i < globalStats.Num; i++ {
+			// GetStatsInfo will return the copy of the statsInfo, so we don't need to worry about the memory problem.
```

Comment:
What "memory problem"? Please be a bit more specific.

Reply:
I have updated. it has been changed as ```data race```.